aiidalab/utils.py

```diff
@@ -214,6 +214,37 @@ def get_package_by_name(packages: dict[str, Package], name: str) -> Package | No
     return None
 
 
+def is_valid_version(version: str) -> bool:
+    """Return True if given string is a PEP440-compliant version, False otherwise."""
+    from packaging.version import InvalidVersion, parse
+
+    try:
+        parse(version)
+    except InvalidVersion:
+        return False
+    else:
+        return True
+
+
+def sort_semantic(
+    versions: Iterable[str], reverse: bool = True, prereleases: bool = False
```

Comment:
I added the `reverse` parameter so that the interface is closer to the builtin `sorted` function, but note that in our case it is by default `True` since we sort in descending order, latest version first.